### PR TITLE
Replace native clients

### PR DESCRIPTION
**Pull request by agocke**
This change continues on the trend of removing the native client code for the compiler server and replacing it with managed code. This is the PR which actually removes the native code! :) This commit deletes all of the native projects and replaces them with the managed client implementation, keeping exactly the same functionality as the old csc2/vbc2 projects.

As before, this PR builds on the commits of the previous one, namely #866, so that one should be reviewed and merged first.

@gafter @jaredpar @AlekseyTs @VladimirReshetnikov @VSadov @pharring

### CODE REVIEW

File: Src/Compilers/CSharp/csc2/Csc2.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using Microsoft.CodeAnalysis.BuildTasks;
```
Comment (pharring) [inline Src/Compilers/CSharp/csc2/Csc2.cs]: Missing OSS header.

File: Src/Compilers/Core/MSBuildTask/BuildClient.cs [hunk 2]
```diff
@@ -0,0 +1,480 @@
+﻿using System;
```
Comment (pharring) [inline Src/Compilers/Core/MSBuildTask/BuildClient.cs]: OSS header required

**Comment by jaredpar** [comment]
Overall looks good.  Left some comments.  Getting hard to track the changes because of the cumulative PR's.  Can you add a ping comment when you update based on feedback?  I'll take another look through then.

**Comment by agocke** [comment]
@jaredpar Sure. And if you want fewer PRs, review faster ;)

**Comment by jaredpar** [comment]
@agocke I need to find a way to sleep less.  That's really the only solution at this point.

**Comment by gafter** [comment]
@agocke I suggest you consider submitting pull requests against your own pull-request branch, thereby enabling reviewers to review only the changes from one review to another, even when both (or all) are pending.

**Comment by agocke** [comment]
@gafter That is a good suggestion. I will do that in the future.

**Comment by agocke** [comment]
Closing for new PR https://github.com/agocke/roslyn/pull/1

**Pull request closed without merging**